lib/assets.js

```diff
@@ -102,17 +112,27 @@ Assets.prototype.serveAsset = function (req, res, next) {
 
   var contentType = asset.contentType;
 
-  if (contentType.match(/^text\/|\/json$|\/javascript$/)) {
-    contentType += "; charset=UTF-8";
-  }
+  if (contentType) {
+    if (contentType.match(/^text\/|\/json$|\/javascript$/)) {
+      contentType += "; charset=UTF-8";
+    }
 
-  res.setHeader("Content-Type", contentType);
+    res.setHeader("Content-Type", contentType);
+  }
   res.setHeader("Content-Length", asset.length);
 
   if (req.method === "HEAD") {
     return res.end();
   }
 
+  if (!asset.buffer) {
+    var localFilePath = pathModule.join(this.options.buildDir, assetName);
+    return fs.readFile(localFilePath, function (err, data) {
+      if (err) return next(err);
+      res.end(data);
+    });
+  }
+
   res.end(asset.buffer);
```

Comment:
If `asset.buffer` isn’t defined, then we’ll be calling `res.end()` twice here — once with no contents, and then a second time when `fs.readFile` returns. 

I’m just going to put this in an `else` for now to avoid that, but it’s a little concerning that this isn’t covered in a test.